src/common/hexchat.c

```diff
@@ -369,6 +369,37 @@ hexchat_lag_check_update (void)   /* this gets called every 0.5 seconds */
 	return 1;
 }
 
+/* call whenever timeout intervals change */
+void
+hexchat_reinit_timers (void)
+{
+	static int lag_check_update_tag = 0;
+	static int lag_check_tag = 0;
+	static int away_tag = 0;
+
+	if (prefs.hex_notify_timeout)
+	{
+		notify_tag = fe_timeout_add_seconds (prefs.hex_notify_timeout,
+						     notify_checklist, NULL);
+	} else
+	{
+		fe_timeout_remove (notify_tag);
+	}
+
+	fe_timeout_remove (away_tag);
+	away_tag = fe_timeout_add_seconds (prefs.hex_away_timeout, away_check, NULL);
+
+	if (prefs.hex_gui_lagometer)
+	{
+		lag_check_update_tag = fe_timeout_add (500, hexchat_lag_check_update, NULL);
```

Comment:
This would result in many timers being created, if there is already a tag it should do nothing, also make sure to set tags to 0 when disabled.